common/lib/dependabot/errors.rb

```diff
@@ -242,6 +242,11 @@ def self.updater_error_details(error)
         "error-type": "git_token_auth_error",
         "error-detail": { message: error.message }
       }
+    when Dependabot::FailedToReplaceEnvInConfig
+      {
```

Comment:
It seems we can reuse the `MissingEnvironmentVariable` error here

Reply:
@abdulapopoola , i had looked into this earlier, issue is that MissingEnvironmentVariable is used by _composer_ (PHP) . _FailedToReplaceEnvInConfig_ (for NPM) and _MissingEnvironmentVariable_ are reporting same kind of error (Env key not provided), But contextually they are different. MissingEnvironmentVariable will output a message "_Could not find a key for xxxx_" while error reported from NPM is _Failed to replace env in config: ${TOKEN}_ . So to keep the format same returned from native helper, I opted to use new error class.

Reply:
The message can be passed in and made configurable per invocation. Tagging @kbukum1 who did something similar.

Reply:
@sachin-sandhu , 

We created a new generic error, `TypedDependabotError`. For your errors can you use that and just return ` error.detail` instead of rewriting since yours is also returning standard error message in the detail? You can check  `IncompatibleCPU`, `NetworkUnsafeHTTP`. Actually I plan to update all defined errors to extend from `TypedDependabotError` later. Just I didn't do it in my PR because it was getting bigger and bigger.

Reply:
Added additional param to _MissingEnvironmentVariable_ so that it can be used to pass exception message to retain the exception information.